node/src/main.rs

```diff
@@ -1,3 +1,72 @@
-fn main() {
-    println!("Hello, world!");
-}
+use crate::network::{P2PNetwork, NetworkCommand, NetworkEvent};
+use crate::storage::ChainState;
+use tracing::{info, error};
+use std::sync::Arc;
+use tokio::sync::mpsc;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Initialize logging
+    tracing_subscriber::fmt::init();
+
+    // Initialize storage
+    let db = Arc::new(storage::BlockchainDB::new("./data")?);
+    let chain_state = ChainState::new(db)?;
+
+    // Initialize network
+    let (mut network, command_tx, mut event_rx) = P2PNetwork::new().await?;
+
+    // Initialize chain sync
+    let mut sync = network::ChainSync::new(chain_state, command_tx.clone());
+
+    // Start network event handling
+    let event_handle = tokio::spawn(async move {
+        while let Some(event) = event_rx.recv().await {
+            match event {
```

Comment:
I think this logic could be rewritten as an event handler that receives an event and is managed as an action.

See similar pattern here example: https://ratatui.rs/templates/component/action-rs/